include/misc/rb.h

```diff
@@ -180,8 +180,8 @@ struct rbnode *_rb_foreach_next(struct rbtree *tree, struct _rb_foreach *f);
  */
 #define RB_FOR_EACH_CONTAINER(tree, node, field)			\
 	for (struct _rb_foreach __f = _RB_FOREACH_INIT(tree, node);	\
-	     (node = CONTAINER_OF(_rb_foreach_next(tree, &__f),		\
-				  __typeof__(*(node)), field));		\
+	     ({node = CONTAINER_OF(_rb_foreach_next(tree, &__f),	\
+			   __typeof__(*(node)), field); node != NULL; }); \
```

Comment:
Agree! and this change even violates other advisory rule, though I didn't see other way to do this in a macro. We can change it to be a function and receive a callback, but this will trigger bigger changes.
I'm open for suggestions here since we have more similar cases.